# Лабораторная работа 3.2. Домашнее задание

### Задание №1

Создайте два массива: в первом должны быть четные числа от 2 до 12 включительно, а в другом числа 7, 11, 15, 18, 23, 29.

$1.$ Сложите массивы и возведите элементы получившегося массива в квадрат:

In [3]:
import numpy as np
a = np.array([2, 4, 6, 8, 10, 12])
b = np.array([7, 11, 15, 18, 23, 29])
print("(a + b)^2:", (a+b)**2)

(a + b)^2: [  81  225  441  676 1089 1681]


$2.$ Выведите все элементы из первого массива, индексы которых соответствуют индексам тех элементов второго массива, которые больше 12 и дают остаток 3 при делении на 5.

In [4]:
print(a[(b > 12) & (b%5==3)])

[ 8 10]


*3.* Проверьте условие "Элементы первого массива делятся на 4, элементы второго массива меньше 14". (Подсказка: в результате должен получиться массив с True и False)

In [7]:
print((a%4==0) & (b<14))

[False  True False False False False]


### Задание №2

* Найдите интересный для вас датасет. Например, можно выбрать датасет тут. http://data.un.org/Explorer.aspx (выбираете датасет, жмете на view data, потом download, выбирайте csv формат)
* Рассчитайте подходящие описательные статистики для признаков объектов в выбранном датасете
* Проанализируйте и прокомментируйте содержательно получившиеся результаты
* Все комментарии оформляйте строго в ячейках формата markdown
    

Для выполнения задания был выбран датасет, по прогнозируемому числу жителей в России до 2100 года.

In [84]:
import csv
import numpy as np

with open('human.csv', newline='') as file:
    reader = csv.DictReader(file)
    data = [row for row in reader]

value = np.array([float(row['Value']) for row in data])
variant = np.array([str(row['Variant']) for row in data])

print(f"Описательные статистики общего числа жителей")
print(f"\nМаксимальное число жителей: ",np.max(value[(variant == "High")]))
print(f"Среднее число жителей: ",round(np.mean(value[(variant == "High")]), 3))
print(f"Минимальное число жителей: ",np.min(value[(variant == "High")]))
print(f"Среднее отклонение числа жителей: ",round(np.std(value[(variant == "High")]), 3))

print(f"\nОписательные статистики рождаемости")
print(f"\nМаксимальная рождаемость: ",np.max(value[(variant == "Constant fertility")]))
print(f"Средняя постоянная рожаемость: ",round(np.mean(value[(variant == "Constant fertility")]), 3))
print(f"Минимальная рождаемость: ",np.min(value[(variant == "Constant fertility")]))
print(f"Среднее отклонение рождаемости: ",round(np.std(value[(variant == "Constant fertility")]), 3))

print(f"\nОписательные статистики смертности")
print(f"\nМаксимальная смертность: ",np.max(value[(variant == "Constant mortality")]))
print(f"Средняя постоянная смертность: ",round(np.mean(value[(variant == "Constant mortality")]), 3))
print(f"Минимальная смертность: ",np.min(value[(variant == "Constant mortality")]))
print(f"Среднее отклонение смертности: ",round(np.std(value[(variant == "Constant mortality")]), 3))


Описательные статистики общего числа жителей

Максимальное число жителей:  162441.155
Среднее число жителей:  147337.055
Минимальное число жителей:  143258.068
Среднее отклонение числа жителей:  5393.453

Описательные статистики рождаемости

Максимальная рождаемость:  144713.314
Средняя постоянная рожаемость:  120630.861
Минимальная рождаемость:  94000.181
Среднее отклонение рождаемости:  15702.513

Описательные статистики смертности

Максимальная смертность:  144713.314
Средняя постоянная смертность:  112912.371
Минимальная смертность:  89460.587
Среднее отклонение смертности:  16587.278


Максимальное количество жителей в России за 100 лет достигнет 162 441 155 человек.

Максимальная рождаемость равна максимальной смертности. отсюда можно сделать вывод, что есть промежуток времени, когда либо смертность будет превышать рождаемость, либо наоборот.

Среднее отклонение числа жителей равна 5 393 453 человека.
Среднее отклонение рождаемости 15 702 513 человек.
Среднее отклонение смертности 16 587 278 человек.